rpm/stackdriver-agent

```diff
@@ -61,71 +60,38 @@ if [ -n "$JAVA_LIB_DIR" ]; then
     export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$JAVA_LIB_DIR"
 fi
 
-get_instance_id () {
-    local iid
-
-    # allow override of instance id in sysconfig file
-    if [ -n "$INSTANCE_ID" ]; then
-        iid=$INSTANCE_ID
-    elif [ -r /opt/stackdriver/hostid ]; then
-        iid=$(cat /opt/stackdriver/hostid)
-    elif [ -z "$SKIP_METADATA_CHECKS" ]; then
-        # GCP: If we're running on GCE, this will return the instance ID.
-        iid=$(curl --silent --connect-timeout 1 -f -H 'Metadata-Flavor: Google' http://169.254.169.254/computeMetadata/v1/instance/id 2>/dev/null)
-        if [ -z "$iid" ]; then
-          # Not running on GCE. Checking AWS.
-          # AWS: If we're on EC2, this ought to return the instance id.
-          iid=$(curl --silent --connect-timeout 1 -f http://169.254.169.254/latest/meta-data/instance-id 2>/dev/null)
-        fi
-    elif [ -r /sys/hypervisor/uuid -a $(cat /sys/hypervisor/version/extra |grep -c amazon) -eq 0 ]; then
-        iid=$(cat /sys/hypervisor/uuid)
-    else
-        echo 'Unable to discover an id for this machine!' >&2
-    fi
-
-    echo $iid
-}
-
 gen_hostid() {
     echo "Generating a host id"
     uuidgen > /opt/stackdriver/hostid
     return 0
 }
 
 # return:
-#   0 if config was generated successfully.
-#   1 if there is a authentication or permissions error.
-gen_config() {
-    if [ "$AUTOGENERATE_COLLECTD_CONFIG" = "no" ]; then
-        return
+#   0 if config is fine
+#   1 if there is a syntax error
+#   2 if there is no configuration
+#   3 if the instance doesn't have the necessary permissions
+check_config() {
+    if test ! -e "$CONFIG"; then
+        return 2
+    fi
+    if ! $DAEMON -t -C "$CONFIG"; then
+        return 1
     fi
-    AUTOGENERATE_HEADER="# NOTE: This file is auto-generated and will be overwritten unless you set
-#     AUTOGENERATE_COLLECTD_CONFIG=\"no\"
-# in /etc/sysconfig/stackdriver
-"
     # Check if the application default credentials file is in the system
     # location.
     if [ ! -f /etc/google/auth/application_default_credentials.json ]; then
         # See if the instance has the correct monitoring scopes.
         INSTANCE_SCOPES=$(curl --silent --connect-timeout 1 -f -H "Metadata-Flavor: Google" http://169.254.169.254/computeMetadata/v1/instance/service-accounts/default/scopes 2>/dev/null || /bin/true)
         if [ `echo "$INSTANCE_SCOPES" | grep -cE '(monitoring.write|monitoring|cloud-platform)$'` -lt 1 ]; then
-            echo >&2 "The instance has neither the application default" \
-              "credentials file nor the correct monitoring scopes; Exiting."
-            return 1
+            echo "The instance has neither the application default" \
```

Comment:
Why did you move the redirection?

Reply:
Sorry, Copy pasted from deb/stackdriver-agent.init. Fixed. 

Is there a difference between the way echo works on debian vs RHEL that affects the positioning?

Reply:
No difference -- having the redirection in the front is a readability thing (makes it obvious where the output is going without having to scroll to the end).